### PR TITLE
First look at faster interactive fmt_

### PR DESCRIPTION
**Pull request by slodge**
# Summary

First trivial attempt at removing the burden of multiple `render_formats` calls for interactive tables.

I've tested this against some local test tables, and I've run the testthat tests - all looks OK... but I'm not sure if this change actually makes sense. I'm really unsure about the partial table build (e.g. no footnotes) that occur inside `extract_cells` - if this is needed then it might be better to split `extract_cells` into two functions?

## Performance
Impact of this PR on:
```
# for local: devtools::load_all("../../play/gt")
library(gt)
library(tidyverse)
library(tictoc)

demo_length <- 10000
my_table <-
  tibble(
    Name = # generate random names
      stringi::stri_rand_strings(demo_length, 10),
    Population = round(runif(demo_length, 100000000, 1000000000)),
    Score = runif(demo_length, 0, 1)
  )

profvis::profvis({
  html1 <- my_table |> 
    gt() |> 
    fmt_number(columns = "Population") |>
    fmt_percent(columns = "Score") |> 
    opt_interactive() |> 
    gt:::as.tags.gt_tbl()
})    
```
is:

| Before PR | After PR |
| ----- | ----- |
| image | image |

# Related GitHub Issues and PRs

- Ref: #1528 

# Checklist

- [x] I understand and agree to the [Code of Conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html).
- [ ] I have listed any major changes in the [NEWS](https://github.com/rstudio/gt/blob/master/NEWS.md).
- [ ] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/rstudio/gt/tree/master/tests/testthat) for any new functionality.

### CODE REVIEW

File: tests/testthat/_snaps/image.new.md
Comment (olivroy) [inline tests/testthat/_snaps/image.new.md:1]: This file should not be committed. 

Could you please delete it? 

If you are not familiar with snapshot tests, find more information at https://testthat.r-lib.org/articles/snapshotting.html

Comment (slodge) [reply tests/testthat/_snaps/image.new.md:1]: Redid the commit and force pushed to remove it completely. 

Sorry - hadn't spotted it (it snuck in when I switched between git clients - different work creds on different clients...)

**Review by rich-iannone — APPROVED** [review]
LGTM!

**Comment by rich-iannone** [comment]
This works great and does make a lot of sense. And footnotes do appear with interactive tables (there's a bug with n == 1 footnotes not appearing in the footer, will fix that next).

If you find any more efficiencies with interactive tables, I'll gladly accept further PRs! (And sorry for taking so long to review this particular PR.)

**Comment by slodge** [comment]
Thanks for merging this - the main other perf issue I found was #1531 - but the only PR I can think of for that is to switch back to base date formatting (which I guess isn't preferred here)

**Pull request merged**